### PR TITLE
New variables and commands for Global Actor

### PR DESCRIPTION
**Pull request by BlurredChoise**
Global Actor (GA) can now save information about time in simulation.
Additionally, GA provides now tool for filtering unwanted processes (for example - save all processes but without transport, etc.).

### CODE REVIEW

File: actors_tools/GlobalActorReader/GlobalActorReader.cc [hunk 1]
```diff
@@ -156,6 +161,16 @@ TVector3 GlobalActorReader::GetPolarizationAfterProcess() {return PolarizationAf
 
 std::string GlobalActorReader::GetProcessName() {return ProcessName.GetValue();}
 
+int GlobalActorReader::GetParentID() {return ParentID.GetValue();}
```
Comment (wkrzemien) [inline actors_tools/GlobalActorReader/GlobalActorReader.cc]: shouldnt be those methods const?
GetParentID() const {return ...}

File: actors_tools/GlobalActorReader/GlobalActorReader.hh [hunk 2]
```diff
@@ -165,6 +175,16 @@ private:
     Variable<TVector3, true> PolarizationAfterProcess;
     //Process name
     Variable<std::string, true> ProcessName;
+    //Parent ID
```
Comment (wkrzemien) [inline actors_tools/GlobalActorReader/GlobalActorReader.hh]: I think in this case those comments are not necessary. They are redundant cause from the variable name you would learn the same

Comment (wkrzemien) [reply actors_tools/GlobalActorReader/GlobalActorReader.hh]: Or you need it for some doxygen ?

File: source/digits_hits/include/GateGlobalActor.hh [hunk 3]
```diff
@@ -235,6 +235,30 @@ public:
...
+	 * */
+	void SetEnableInteractionTime();
+
+	/** Set enable local time.
+	    Please remember that definition of local time is: time since the track was created.
+	 * */
+	void SetEnableLocalTime();
+
+	/** Set enable global time.
+	    Please remember that definition of global time is: time since the event was created.
```
Comment (wkrzemien) [inline source/digits_hits/include/GateGlobalActor.hh]: so what is the difference between this and 248?

File: source/digits_hits/include/GateGlobalActor.hh [hunk 4]
```diff
@@ -309,6 +333,26 @@ private:
...
 
+	/** Update data about parent ID
+	 * */
+	void UpdateParentID(const G4Step& step);
+
+	/** Update interaction time value
+	 * */
+	void UpdateInteractionTime(const G4Step& step);
+
+	/** Update local time
```
Comment (wkrzemien) [inline source/digits_hits/include/GateGlobalActor.hh]: I really think those comment are redunand, since you chose good names for your functions

Comment (wkrzemien) [reply source/digits_hits/include/GateGlobalActor.hh]: http://www.informit.com/articles/article.aspx?p=1327761

File: actors_tools/GlobalActorReader/GlobalActorReader.cc (hunk 1 above)
Comment (BlurredChoise) [reply actors_tools/GlobalActorReader/GlobalActorReader.cc]: Done

File: actors_tools/GlobalActorReader/GlobalActorReader.hh (hunk 2 above)
Comment (BlurredChoise) [reply actors_tools/GlobalActorReader/GlobalActorReader.hh]: I will extend description of this variables - this for people who don't know what specific variable does and means. Moreover I think about creating doxygen documentation in nearest future.

File: source/digits_hits/include/GateGlobalActor.hh (hunk 4 above)
Comment (BlurredChoise) [reply source/digits_hits/include/GateGlobalActor.hh]: I agree. I will remove methods descriptions, but leave variables descriptions - in my opinion variables descriptions are critical for code understanding.

File: source/digits_hits/include/GateGlobalActor.hh (hunk 3 above)
Comment (BlurredChoise) [reply source/digits_hits/include/GateGlobalActor.hh]: My mistake - interaction time is local time difference between post and pre step. Description changed.

**Comment by BlurredChoise** [comment]
Code ready for next PR iteration.

File: actors_tools/GlobalActorReader/GlobalActorReader.hh [hunk 5]
```diff
@@ -82,89 +82,109 @@ private:
...
+
+    double GetProperTime() const;
 
 private:
 /** VARIABLES
  * Declar here your variable by using template class Variable<yourType>
  * Please add description of your variable here
 **/
-    //Volume name (layer name)
+    //Volume name (layer name) - this varaible tell in which volume (layer) something happened e.g. gamma scattered in volume wit name "LayerA"
```
Comment (wkrzemien) [inline actors_tools/GlobalActorReader/GlobalActorReader.hh]: varaible -> variable

File: actors_tools/GlobalActorReader/GlobalActorReader.hh [hunk 6]
```diff
@@ -82,89 +82,109 @@ private:
...
     Variable<std::string, true> VolumeName;
-    //Scintilator translation vector (if detector does not rotate it is equal scintilator centrum position
+    //Scintilator translation vector (if detector does not rotate it is equal scintilator centrum position) - this is useful when you want to know where exacly process happened (from VolumeName you know layer and from ScintilatorPosition you know which scintilator)
     Variable<TVector3, true> ScintilatorPosition;
-    //Event ID
+    //Event ID - this variable tell you which event is related to particl e.g. you have e+ and gamma - both created during event no 1 (both has the same EventID)
```
Comment (wkrzemien) [inline actors_tools/GlobalActorReader/GlobalActorReader.hh]: particle

File: actors_tools/GlobalActorReader/GlobalActorReader.hh [hunk 7]
```diff
@@ -82,89 +82,109 @@ private:
...
     Variable<double, false> EnergyLossDuringProcess;
     //Particle momemntum direction before process (e.g. before scattering)
     Variable<TVector3, true> MomentumDirectionBeforeProcess;
     //Particle momemntum direction after process (e.g. after scattering)
     Variable<TVector3, true> MomentumDirectionAfterProcess;
-    //Process position in lab coordinate system
+    //Process position in lab coordinate system - by this variable you know where exacly process happened
     Variable<TVector3, true> ProcessPosition;
-    //Emission point from source
+    //Emission point from source - by this varaible you know position of source which emitted particle
```
Comment (wkrzemien) [inline actors_tools/GlobalActorReader/GlobalActorReader.hh]: variable

**Comment by BlurredChoise** [comment]
I added corrections to the descriptions

**Pull request merged**